### ISSUE TITLE
expiring-alerts: Can be run from command line but errors from CRON

### ISSUE THREAD

**Issue by ghost** [snipe/snipe-it#3937]
Hi,

When adding the following to CRON:
* * * * * /usr/bin/php /var/www/html/artisan snipeit:expiring-alerts > /tmp/snipeit

Expecting the expiring assets and licenses to send e-mail every minute. (For testing, plan on making it every two weeks though)

The errors in the /tmp/snipeit file are:

 [PDOException]
  SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: Temporary failure in name resolution

  [PDOException]
  PDO::__construct(): php_network_getaddresses: getaddrinfo failed: Temporary failure in name resolution

I can run it from the command line like this:
/usr/bin/php /var/www/html/artisan snipeit:expiring-alerts

and it sends the e-mails and I get them successfully
1 expiring assets
1 expiring licenses
Report sent to tdmcfall@gmail.com
Report sent to tdmcfall@gmail.com

I have tried many attempts at solutions including, but not limited to, making it an executable script (making sure bash was the shell), using cd /var/www/html/ in CRON, using the unlinked PHP file of /usr/bin/php7.0, etc., but I continued to get the same errors. However, I could still run it throughout my troubleshooting successfully straight from the command line.

I am using a docker image and it is the latest tag. Created 11 days ago as of this writing.

I didn't enable debug mode, but I did look in the laravel.log and couldn't find anything. I am not very versed in Laravel though.

Version of Snipe-IT is 3.6.6 build 1

I had no errors in the installation

I have not modified the database directly

I am able to use all other aspects of the app successfully.

Kudos for this app!

Thanks!
Troy

**Comment by snipe** [comment]
That looks like a server configuration issue, not a Snipe-IT issue. 

Possible reasons:

- the cron user has `nologin` as their shell. See https://github.com/piwik/docker-piwik/issues/17#issuecomment-179924825 for a potential solution.
- [This (old) PHP issue](https://serverfault.com/a/16021).

**Comment by uberbrady** [comment]
The only thing I can think of is your 'current directory' might be hard for cron's shell to figure out?

Maybe try something this as your command in cron, instead?

```
cd /var/www/html; /usr/bin/php artisan snipeit:expiring-alerts
```

**Comment by troy-mcfall** [comment]
@snipe,
Maybe I don't know Linux well enough, but I am using the root users CRON. On the snipe/snipe-IT latest docker container, I had to install CRON to be able to use CRON and crontab -e to create one. I even tried  running a script in CRON which had bash specifically identified. I could run it from the command line successfully, but same errors when run from CRON.
On the old PHP issue, I wasn't sure which service to restart. httpd didn't exist and when I tried Apache2, the docker container stopped completely. Did I misunderstand the fix for that issue?

@uberbrady,
I had tried that method in CRON previously as well, but I didn't use your exact syntax, so tried it and still get the same errors.


From what I have found on the errors, it could be some sort of DNS lookup issue? The only modification I have done to the latest version of the snipe-IT running container is install CRON. The /etc/resolv.conf and /etc/hosts files are readable by everyone and they are correct. I haven't had any issues with the application other than this one so far.

Is there anything I can specifically provide (files, screens, etc.) to you to help in better understanding what I am doing wrong?

Thanks!
Troy
